nft/ethereum_payment_obligation.go

```diff
@@ -247,6 +257,25 @@ func (s *ethereumPaymentObligation) minter(ctx context.Context, tokenID TokenID,
 	}
 }
 
+func isNFTMinted(nfts []*coredocumentpb.NFT, registry []byte) bool {
```

Comment:
Should we maybe check also if the NFT exists in the registry itself?

What would happen, if I want to mint an NFT  and the document update is successful but 
the Ethereum transaction fails?

It would not be possible to retry the minting because `isNFTMinted` would return true?

Maybe isNFTMinted should only return true if the registry is in the CD and the tokenID exists in the contract?

Reply:
That is not that simple. If we want to check the contract, we first need to connect document with NFT. Then verify if an NFT has been minted for this document through this registry.

I am a little out of sync with the smart contracts at the moment, so if we have a way to check this, then we can check that. But if we have that functionality, then we dont need to have this as a part of coredocument.

Reply:
The method is in ethereumPaymentObligation
```golang
 func (s *ethereumPaymentObligation) OwnerOf(registry common.Address, tokenID []byte (owner common.Address, err error) 
```
The check would basically be `if address != "0x000000..."` the tokenID is minted.

We need to store it in the coredocument as well. Because in the future an additional proof is needed, that the tokenID/registry is part of the coredocument to proof uniqueness. 

But maybe we don't need the addtional check for now in the isNFTMinted method

Reply:
Nice!
Let me update.